tests/integration/workflows/nodejs_npm/test_nodejs_npm.py

```diff
@@ -43,6 +48,23 @@ def test_builds_project_without_dependencies(self):
         output_files = set(os.listdir(self.artifacts_dir))
         self.assertEqual(expected_files, output_files)
 
+    def test_builds_project_without_manifest(self):
+        source_dir = os.path.join(self.TEST_DATA_FOLDER, "no-manifest")
+
+        with mock.patch.object(logger, "warning") as mock_warning:
```

Comment:
Why mock the logger?

Reply:
Just to validate that the appropriate warning was logged. Is there a better way to do this?

Reply:
Ooh, wasn't aware `with` doesn't start a new scope! Just noticed the `mock_warning` assertion.